services/controllerhost/api_handlers.go

```diff
@@ -353,6 +357,13 @@ func refreshInputHostsForDst(context *Context, dstUUID string, now int64) ([]str
 	}
 
 	var dstType = getDstType(dstDesc)
+
+	// Fail attempts to publish to Kafka destinations
+	if dstType == dstTypeKafka {
```

Comment:
>  I'd rather update the cache to support and remember to return an "error" on the request.
That's cool and even better.

> " optimizing for very uncommon case"
I am ok with punting this on v1, but our system needs to be robust against bad software/config mistakenly ddos'ing us. That actually happens more commonly than actual ddos attacks, in practice :) Just to be clear, this is not an optimization, it's about robustness; its about not browning out the metadata (on bad input). 

leaving it up to you to decide if you want to take up this now or later.